src/devices/src/virtio/block/persist.rs

```diff
@@ -89,6 +89,11 @@ impl Persist<'_> for Block {
     type Error = io::Error;
 
     fn save(&self) -> Self::State {
+        // Sync data out to backing file on host.
+        if self.disk.file().sync_all().is_err() {
```

Comment:
Should we also flush first ? In order to force any rust cached data out to the OS caches ?

Reply:
Good catch Serban! Currently `flush` is noop in rust stdlib, but we should definitely do it explicitly instead of relying on underlying mechanisms which could change in the future.